plugins/destination/mongodb/client/write.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func (c *Client) appendTableBatch(ctx context.Context, table *schema.Table, resources [][]any) error {
+	documents := make([]any, len(resources))
+	for i, resource := range resources {
+		document := make(bson.M, len(table.Columns))
+		for l, col := range table.Columns {
+			document[col.Name] = resource[l]
+		}
+		documents[i] = document
+	}
+	if _, err := c.client.Database(c.pluginSpec.Database).Collection(table.Name).InsertMany(ctx, documents); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Client) overwriteTableBatch(ctx context.Context, table *schema.Table, resources [][]any) error {
+	operations := make([]mongo.WriteModel, len(resources))
+	pks := table.PrimaryKeys()
+	for i, resource := range resources {
+		operation := mongo.NewUpdateOneModel()
+		operation.SetUpsert(true)
+		if c.spec.WriteMode == specs.WriteModeOverwrite || c.spec.WriteMode == specs.WriteModeOverwriteDeleteStale {
```

Comment:
I think this `if` statement is redundant, because `overwriteTableBatch` should only be called if this is already true? (As is done in `WriteTableBatch`)

Reply:
Yeah good catch. forgot to remove this.